component/azstorage/block_blob.go

```diff
@@ -606,10 +607,35 @@ func (bb *BlockBlob) WriteFromFile(name string, metadata map[string]string, fi *
 	//defer exectime.StatTimeCurrentBlock("WriteFromFile::WriteFromFile")()
 
 	blobURL := bb.Container.NewBlockBlobURL(filepath.Join(bb.Config.prefixPath, name))
-
 	defer log.TimeTrack(time.Now(), "BlockBlob::WriteFromFile", name)
+	blockSize := bb.Config.blockSize
+	// if the block size is not set then we configure it based on file size
+	if !config.IsSet(compName + ".block-size-mb") {
+		// get the size of the file
+		stat, err := fi.Stat()
+		if err != nil {
+			log.Err("BlockBlob::WriteFromFile : Failed to get file size %s (%s)", name, err.Error())
+			return err
+		}
+		fileSize := stat.Size()
+		// If bufferSize > (BlockBlobMaxStageBlockBytes * BlockBlobMaxBlocks), then error
+		if fileSize > azblob.BlockBlobMaxStageBlockBytes*azblob.BlockBlobMaxBlocks {
+			err = errors.New("buffer is too large to upload to a block blob")
+			log.Err("BlockBlob::WriteFromFile : buffer is too large to upload to a block blob %s", name)
```

Comment:
Shall we say "block blob" or just "blob".

Reply:
I think block blob is more accurate and informative

Reply:
block blob is fine I think.